// @salesforce/design-system-react/module/utilities/constants.js
// ## Components
export var ACCORDION = 'SLDSAccordion';
export var ACCORDION_PANEL = 'SLDSAccordionPanel';
export var ALERT = 'SLDSAlert';
export var ALERT_CONTAINER = 'SLDSAlertContainer';
export var APP_LAUNCHER = 'SLDSAppLauncher';
export var APP_LAUNCHER_SECTION = 'SLDSAppLauncherSection';
export var APP_LAUNCHER_TILE = 'SLDSAppLauncherTile';
export var APP_LAUNCHER_HEADER = 'SLDSAppLauncherHeader';
export var AVATAR = 'SLDSAvatar';
/* Remove at next breaking change */

export var BREAD_CRUMB = 'SLDSBreadcrumb';
export var BREADCRUMB = 'SLDSBreadcrumb';
export var BUTTON = 'SLDSButton';
export var BUTTON_GROUP = 'SLDSButtonGroup';
export var BUTTON_STATEFUL = 'SLDSButtonStateful';
export var BUTTON_ICON = 'SLDSButtonIcon'; // a non-interctive icon wrapper for SLDSUtilityIcon

export var CARD = 'SLDSCard';
export var CARD_BODY = 'SLDSCardBody';
export var CARD_EMPTY = 'SLDSCardEmpty';
export var CARD_FILTER = 'SLDSCardFilter';
export var CARD_FOOTER = 'SLDSCardFooter';
export var CARD_HEADER = 'SLDSCardHeader';
export var COMBOBOX = 'SLDSCombobox';
export var DATA_TABLE = 'SLDSDataTable';
export var DATA_TABLE_CELL = 'SLDSDataTableCell';
export var DATA_TABLE_COLUMN = 'SLDSDataTableColumn';
export var DATA_TABLE_HEAD = 'SLDSDataTableHead';
export var DATA_TABLE_HEADER_CELL = 'SLDSDataTableHeaderCell';
export var DATA_TABLE_ROW_ACTIONS = 'SLDSDataTableRowActions';
export var DATA_TABLE_ROW = 'SLDSDataTableRow';
export var DATE_PICKER = 'SLDSDatePicker';
export var DIALOG = 'SLDSDialog';
export var FILTER = 'SLDSFilter';
export var FORMS_CHECKBOX = 'SLDSCheckbox';
export var FORMS_INLINE_EDIT = 'SLDSInlineEdit';
export var FORMS_INPUT = 'SLDSInput';
export var FORMS_TEXTAREA = 'SLDSTextarea';
export var FORMS_SEARCH = 'SLDSSearch';
export var GLOBAL_HEADER = 'SLDSGlobalHeader';
export var GLOBAL_HEADER_PROFILE = 'SLDSGlobalHeaderProfile';
export var GLOBAL_HEADER_SEARCH = 'SLDSGlobalHeaderSearch';
export var GLOBAL_HEADER_TOOL = 'SLDSGlobalHeaderTool';
export var GLOBAL_NAVIGATION_BAR = 'SLDSGlobalNavigationBar';
export var GLOBAL_NAVIGATION_BAR_BUTTON = 'SLDSGlobalNavigationBarButton';
export var GLOBAL_NAVIGATION_BAR_DROPDOWN = 'SLDSGlobalNavigationBarDropdown';
export var GLOBAL_NAVIGATION_BAR_LABEL = 'SLDSGlobalNavigationBarLabel';
export var GLOBAL_NAVIGATION_BAR_LINK = 'SLDSGlobalNavigationBarLink';
export var GLOBAL_NAVIGATION_BAR_REGION = 'SLDSGlobalNavigationBarRegion';
export var GLOBAL_NAVIGATION_BAR_APP_LAUNCHER = 'SLDSGlobalNavigationBarAppLauncher';
export var GRID = 'SLDSGrid';
export var HIGHLIGHTER = 'SLDSHighlighter';
export var ICON = 'SLDSIcon';
export var ICON_SETTINGS = 'SLDSIconSettings';
export var ICON_INPUT = 'SLDSIconInput';
export var LIST = 'SLDSList';
export var LIST_ITEM = 'SLDSListItem';
export var LIST_ITEM_LABEL = 'SLDSListItemLabel';
export var LOOKUP = 'SLDSLookup';
export var MEDIA_OBJECT = 'SLDSMediaObject';
export var MENU_DROPDOWN = 'SLDSMenuDropdown';
export var MENU_DROPDOWN_TRIGGER = 'SLDSMenuDropdownTrigger';
export var MENU_PICKLIST = 'SLDSMenuPicklist';
export var MODAL = 'SLDSModal';
export var NAVIGATION = 'SLDSNavigation';
export var NAVIGATION_ITEM = 'SLDSNavigationItem';
export var NOTIFICATION = 'SLDSNotification';
export var PANEL = 'SLDSPanel';
export var PANEL_FILTER_GROUP = 'SLDSFilterGroup';
export var PANEL_FILTER_LIST = 'SLDSFilterList';
export var PANEL_FILTER_LIST_HEADING = 'SLDSPanelFilterListHeading';
export var PAGE_HEADER = 'SLDSPageHeader';
export var PILL = 'SLDSPill';
export var POPOVER = 'SLDSPopover';
export var POPOVER_TOOLTIP = 'SLDSPopoverTooltip';
export var PROGRESS_INDICATOR = 'SLDSProgressIndicator';
export var PROGRESS_INDICATOR_PROGRESS = 'SLDSProgressIndicatorProgress';
export var PROGRESS_INDICATOR_STEP = 'SLDSProgressIndicatorStep';
export var PROGRESS_RING = 'SLDSProgressRing';
export var SLIDER = 'SLDSSlider';
export var SPINNER = 'SLDSSpinner';
export var SPLIT_VIEW = 'SLDSSplitView';
export var SPLIT_VIEW_HEADER = 'SLDSSplitViewHeader';
export var SPLIT_VIEW_LISTBOX = 'SLDSSplitViewListbox';
export var RADIO = 'SLDSRadio';
export var RADIO_GROUP = 'SLDSRadioGroup';
export var RADIO_BUTTON_GROUP = 'SLDSRadioButtonGroup';
export var TAB = 'SLDSTab';
export var TAB_ITEM = 'SLDSTabItem';
export var TAB_PANEL = 'SLDSTabPanel';
export var TABS = 'SLDSTabs';
export var TABS_LIST = 'SLDSTabsList';
export var TABS_PANEL = 'SLDSTabsPanel';
export var TIME_PICKER = 'SLDSTimepicker';
export var TOAST = 'SLDSToast';
export var TOAST_CONTAINER = 'SLDSToastContainer';
export var TREE = 'SLDSTree';
export var TREE_BRANCH = 'SLDSTreeBranch';
export var TREE_ITEM = 'SLDSTreeItem';
